domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -34,17 +36,31 @@ public async Task<ActionResult> Create([FromBody] CreateTransferAgreement reques
         var subjectName = User.FindSubjectNameClaim();
         var subjectTin = User.FindSubjectTinClaim();
 
+        var validator = new CreateTransferAgreementValidator(subjectTin);
+
+        var validateResult = await validator.ValidateAsync(request);
+        if (!validateResult.IsValid)
+        {
+            validateResult.AddToModelState(ModelState);
+            return ValidationProblem(ModelState);
+        }
+
         var transferAgreement = new TransferAgreement
         {
             StartDate = DateTimeOffset.FromUnixTimeSeconds(request.StartDate),
-            EndDate = DateTimeOffset.FromUnixTimeSeconds(request.EndDate),
+            EndDate = request.EndDate.HasValue ? DateTimeOffset.FromUnixTimeSeconds(request.EndDate.Value) : null,
             ActorId = actor,
             SenderId = Guid.Parse(subject),
             SenderName = subjectName,
             SenderTin = subjectTin,
             ReceiverTin = request.ReceiverTin
         };
 
+        if (await transferAgreementRepository.HasDateOverlap(transferAgreement))
+        {
+            return Conflict("Transfer agreement date overlap");
```

Comment:
If we were to align with the cert domain we should modify the message of the conflict. So i would prefer if we just return `Conflict();`